app/src/main/java/org/sil/gatherwords/EntryActivity.java

```diff
@@ -297,4 +310,35 @@ public PlayButton(Context ctx) {
         }
     }
 
+
+    private static class AddNewWordToDB extends AsyncTask<Void, Void, Void> {
+        WeakReference<EntryActivity> entryActivityRef;
+        WordDao wDAO;
+        Long sessionID;
+
+        AddNewWordToDB(EntryActivity entryActivity) {
+            entryActivityRef = new WeakReference<>(entryActivity);
+            wDAO = AppDatabase.get(entryActivity).wordDao();
+            sessionID = entryActivity.sessionID;
+        }
+
+        @Override
+        protected Void doInBackground(Void... v) {
+            Word newWord = new Word();
+            newWord.sessionID = sessionID;
+            wDAO.insertWord(newWord);
+            return null;
+        }
+
+        @Override
+        protected void onPostExecute(Void v) {
+            final EntryActivity entryActivity = entryActivityRef.get();
+            if ( entryActivity == null ) {
+                return;
+            }
+            ViewPager pager = entryActivity.findViewById(R.id.viewpager);
+            new LoadWordIDsTask((EntryPagerAdapter)pager.getAdapter(), sessionID, wDAO).execute();
+            return;
```

Comment:
No need to `return` from a `void` function.

Reply:
Fixed.